### PR TITLE
Phase 6: MC Dropout in Surface Refine Head — Stochastic Inference

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Add dropout (p=0.05) inside the SurfaceRefinementHead, then at inference time run K=8 stochastic forward passes and average predictions. This approximates Bayesian model averaging at near-zero training overhead. Different from our existing ensemble (which requires 8x training cost) — MC Dropout gets ensemble-like variance reduction from a SINGLE trained model.

## Instructions

**Step 1 — Add dropout to SurfaceRefinementHead:**
In train.py, locate the `SurfaceRefinementHead` class. Add `nn.Dropout(p)` after each hidden layer's activation (before the next linear layer). Add a CLI flag `--mc_dropout_p` (default 0.0 = disabled).

\`\`\`python
# In config dataclass:
mc_dropout_p: float = 0.0    # MC Dropout probability in surface refine head (0=disabled)
mc_dropout_K: int = 8        # Number of stochastic forward passes at inference

# In SurfaceRefinementHead, after each ReLU/GELU activation:
if self.dropout_p > 0:
    x = F.dropout(x, p=self.dropout_p, training=True)  # ALWAYS True (MC Dropout)
\`\`\`

IMPORTANT: Use `training=True` even at inference time — this is what makes MC Dropout work. The backbone should NOT use dropout, only the surface refinement head.

**Step 2 — Add MC averaging at validation time:**
In the validation loop, when mc_dropout_K > 1 and mc_dropout_p > 0:
- Run K forward passes with dropout active
- Average the K predictions
- Report both single-pass and K-averaged MAE

**Step 3 — Run 8 experiments:**

| GPU | wandb_name | mc_dropout_p | mc_dropout_K | seed |
|-----|-----------|-------------|-------------|------|
| 0 | nezuko/baseline-s42 | 0.0 | 1 | 42 |
| 1 | nezuko/baseline-s43 | 0.0 | 1 | 43 |
| 2 | nezuko/mcd-p0.03-s42 | 0.03 | 8 | 42 |
| 3 | nezuko/mcd-p0.03-s43 | 0.03 | 8 | 43 |
| 4 | nezuko/mcd-p0.05-s42 | 0.05 | 8 | 42 |
| 5 | nezuko/mcd-p0.05-s43 | 0.05 | 8 | 43 |
| 6 | nezuko/mcd-p0.1-s42 | 0.10 | 8 | 42 |
| 7 | nezuko/mcd-p0.1-s43 | 0.10 | 8 | 43 |

Full baseline command (add mc_dropout flags to each):
\`\`\`bash
python train.py --agent nezuko --wandb_group phase6/mc-dropout \
  --asinh_pressure --asinh_scale 0.75 --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --aug aoa_perturb --aug_full_dsdf_rot --high_p_clamp --n_layers 3 --slice_num 96 \
  --tandem_ramp --domain_layernorm --domain_velhead --ema_decay 0.999 --weight_decay 5e-5 \
  --cosine_T_max 160 --disable_pcgrad --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --surface_refine_hidden 192 --surface_refine_layers 3
\`\`\`

**What to report:**
- Surface MAE (p_in, p_oodc, p_tan, p_re) for each config at K=1 and K=8
- Does MC averaging (K=8) improve over single pass (K=1)?
- Which dropout rate works best?

## Baseline
Single-model 8-seed mean: p_in=13.03±0.39, p_oodc=7.83±0.19, p_tan=30.29±0.47, p_re=6.45±0.05
8-seed ensemble: p_in=12.4, p_oodc=6.7, p_tan=29.4, p_re=5.8
W&B: wandb-applied-ai-team/senpai-v1

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2088 because ADVISOR: Closing — MC Dropout is a null result.

**W&B analysis (phase6/mc-dropout, 8 finished runs):**

| Config | p_in | p_oodc | p_tan | p_re |
|--------|------|--------|-------|------|
| baseline | 13.35 | 7.95 | 31.31 | 6.61 |
| mc-p0.05-k16 | 13.29 (-0.4%) | 8.07 (+1.5%) | 31.34 (flat) | 6.59 (flat) |
| mc-p0.05-k8 | 13.73 (+2.8%) | 7.94 (flat) | 32.06 (+2.4%) | 6.46 (-2.3%) |
| mc-p0.10-k8 | 13.26 (-0.7%) | 7.70 (-3.1%) | 32.41 (+3.5%) | 6.38 (-3.5%) |

No consistent improvement. With p=0.05, MC averaging (K=8 or K=16) produces results indistinguishable from baseline. With p=0.10, there's a tradeoff (better p_oodc/p_re but worse p_tan) rather than a free improvement.

The dropout-induced stochasticity in the surface refine head doesn't create enough meaningful diversity to improve predictions via averaging. This is consistent with our finding that the Transolver+EMA training already provides strong implicit regularization.

Reassigning you to a new experiment..
